source/slang/slang-check-decl.cpp

```diff
@@ -7261,12 +7307,172 @@ namespace Slang
+    
+        auto structDecl = as<StructDecl>(aggTypeDecl);
+        if (!structDecl)
+            return;
+
+        struct DeclAndCtorInfo
+        {
+            StructDecl* parent = nullptr;
+            ConstructorDecl* defaultCtor = nullptr;
+            List<ConstructorDecl*> ctorList;
+            DeclAndCtorInfo()
+            {
+            }
+            DeclAndCtorInfo(StructDecl* parent, const bool getOnlyDefault)
+            {
+                if (getOnlyDefault)
+                    defaultCtor = _getDefaultCtor(parent);
+                else
+                    ctorList = _getCtorList(parent, &defaultCtor);
+            }
+        };
+        List<DeclAndCtorInfo> inheritanceDefaultCtorList{};
+        for (auto member : structDecl->members)
+        {
+            auto inheritanceMember = as<InheritanceDecl>(member);
+            if (!inheritanceMember)
+                continue;
+            auto declRefType = as<DeclRefType>(inheritanceMember->base.type);
+            if (!declRefType)
+                continue;
+            auto structOfInheritance = as<StructDecl>(declRefType->getDeclRef().getDecl());
+            if (!structOfInheritance)
+                continue;
+            inheritanceDefaultCtorList.add(DeclAndCtorInfo(structOfInheritance, true));
+        }
+        DeclAndCtorInfo structDeclInfo = DeclAndCtorInfo(structDecl, false);
+
+        std::size_t insertOffset = 0;
+        Dictionary<Decl*, Expr*> cachedDeclToCheckedVar;
+        for (auto& declInfo : inheritanceDefaultCtorList)
+        {
+            if (!declInfo.defaultCtor)
+                continue;
+            for (auto ctor : structDeclInfo.ctorList)
+            {
+                auto seqStmt = _ensureCtorBodyIsSeqStmt(m_astBuilder, ctor);
+                auto ctorName = declInfo.defaultCtor->getName();
+
+                auto lookupResult = lookUpMember(
+                    m_astBuilder,
+                    this,
+                    ctorName,
+                    declInfo.defaultCtor->returnType.type,
+                    ctor->ownedScope,
+                    LookupMask::Function,
+                    LookupOptions::IgnoreInheritance);
+
+                auto invoke = m_astBuilder->create<InvokeExpr>();
+                invoke->functionExpr = createLookupResultExpr(ctorName, lookupResult, nullptr, ctor->loc, nullptr);
+
+                ThisExpr* thisExpr = m_astBuilder->create<ThisExpr>();
+                thisExpr->scope = ctor->ownedScope;
+                thisExpr->type = ctor->returnType.type;
+
+                auto assign = m_astBuilder->create<AssignExpr>();
+                assign->left = coerce(CoercionSite::Initializer, declInfo.defaultCtor->returnType.type, thisExpr);
+                assign->right = invoke;
+                auto stmt = m_astBuilder->create<ExpressionStmt>();
+                stmt->expression = assign;
+                stmt->loc = ctor->loc;
+
+                seqStmt->stmts.insert(insertOffset, stmt);
```

Comment:
Avoid insert this many times. Create a single SeqStmt to hold all new stmts and simply insert that seqStmt at top.

Reply:
I will do the following for the members since it will reduce inserts by N, where N is number of init expression members)

I think we should not do the following for init expressions of parents since we only insert 0-1 statements per function. We are going to have to allocate one SeqStmt+N-loop-of-elements regardless.